### PR TITLE
[FEATURE] Add lifecycle and build pipelines in spk setup command

### LINKED ISSUE #1169: Create LifeCycle Pipeline in spk setup command
**As a:**
User 

**I want:**
LifeCycle Pipeline created for me

**So that:**
Can understand how to use SPK with less manual setup

**Does this require updates to documentation?:**
YES

Related to #986

### PR DESCRIPTION
**Pull request by dennisseah**
and a bunch of fixes from Andre to get the pipelines to work properly.
closes https://github.com/microsoft/bedrock/issues/1169

### CODE REVIEW

**Review by andrebriggs — APPROVED** [review]
LGTM

File: src/commands/setup.md [hunk 1]
```diff
@@ -16,11 +16,9 @@ for a few questions
 4. To create a sample application Repo
    1. If Yes, a Azure Service Principal is needed. You have 2 options
       1. have the command line tool to create it. Azure command line tool shall
-         be used
-      2. provide the Service Principal Id, Password and Tenant Id.
-   2. Subscription Id is automatically retrieved with the Service Principal
-      credential. In case, there are two or more subscriptions, you will be
-      prompt to select one of them.
+         be used. You will be prompt to select a subscription identifier
```
Comment (bnookala) [inline src/commands/setup.md]: Wording: "You will be prompted to select a subscription identifier."

File: src/commands/setup.md [hunk 2]
```diff
@@ -16,11 +16,9 @@ for a few questions
...
+      2. provide the Service Principal Id, Password and Tenant Id. From these
```
Comment (bnookala) [inline src/commands/setup.md]: Wording: "Provide the Service Principal Id, Password, and Tenant Id. From this information, the tool will retrieve the subscription identifier."

File: src/commands/setup.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ The followings shall be created
       already exists.
    5. A Git Repo, `quick-start-app`, it shall be deleted and recreated if is
       already exists.
+   6. A Life Cycle pipeline.
```
Comment (bnookala) [inline src/commands/setup.md]: Lifecycle pipeline

File: src/lib/azure/containerRegistryService.ts [hunk 4]
```diff
@@ -131,13 +157,12 @@ export const create = async (
...
     servicePrincipalTenantId,
     subscriptionId,
-    resourceGroup,
     name
   );
 
   if (exist) {
     logger.info(
-      `Azure container registry, ${name} in ${resourceGroup} already existed`
+      `Azure container registry, ${name} already existed in subscription`
```
Comment (bnookala) [inline src/lib/azure/containerRegistryService.ts]: "already exists in subscription"

File: src/lib/azure/servicePrincipalService.test.ts [hunk 5]
```diff
@@ -22,20 +26,14 @@ describe("test createWithAzCLI function", () => {
...
+    const sub = await createWithAzCLI("subscriptionId");
+    expect(sub.id).toBe(result.appId);
+    expect(sub.password).toBe(result.password);
+    expect(sub.tenantId).toBe(result.tenant);
   });
   it("negative test", async () => {
-    jest.spyOn(servicePrincipalService, "azCLILogin").mockResolvedValueOnce();
     jest.spyOn(shell, "exec").mockRejectedValueOnce(Error("fake"));
-    await expect(createWithAzCLI()).rejects.toThrow();
+    await expect(createWithAzCLI("subscfriptionId")).rejects.toThrow();
```
Comment (bnookala) [inline src/lib/azure/servicePrincipalService.test.ts]: typo: should be "subscriptionId"

File: src/lib/setup/pipelineService.ts [hunk 6]
```diff
@@ -149,6 +162,22 @@ export const pollForPipelineStatus = async (
...
+  rc: RequestContext,
+  pipelineName: string
+): Promise<void> => {
+  const pipeline = await getPipelineByName(
+    buildApi,
+    rc.projectName,
+    pipelineName
+  );
+  if (pipeline && pipeline.id) {
+    logger.info(`${pipelineName} is found, deleting it`);
```
Comment (bnookala) [inline src/lib/setup/pipelineService.ts]: Should make explicit that the pipeline named `${pipelineName}` was found:

logger.info(`Pipeline ${pipelineName} was found - deleting pipeline`);

File: src/lib/setup/prompt.ts [hunk 7]
```diff
@@ -237,3 +263,23 @@ export const getAnswerFromFile = (file: string): RequestContext => {
...
+): Promise<boolean> => {
+  const urlPR = `${getAzureRepoUrl(
+    rc.orgName,
+    rc.projectName,
+    HLD_REPO
+  )}/pullrequest`;
+  const questions = [
+    {
+      default: true,
+      message: `Please approve and merge a PR at ${urlPR}? Refresh the page after a while if you do not see active PR.`,
```
Comment (bnookala) [inline src/lib/setup/prompt.ts]: "Please approve and merge the Pull Request at ${urlPR}? Refresh the page if you do not see an active Pull Request."

**Review by bnookala — CHANGES_REQUESTED** [review]

File: src/commands/setup.md (hunk 1 above)
Comment (dennisseah) [reply src/commands/setup.md]: accept

File: src/commands/setup.md (hunk 2 above)
Comment (dennisseah) [reply src/commands/setup.md]: accept

File: src/commands/setup.md (hunk 3 above)
Comment (dennisseah) [reply src/commands/setup.md]: accept

File: src/lib/azure/containerRegistryService.ts (hunk 4 above)
Comment (dennisseah) [reply src/lib/azure/containerRegistryService.ts]: accept

File: src/lib/azure/servicePrincipalService.test.ts (hunk 5 above)
Comment (dennisseah) [reply src/lib/azure/servicePrincipalService.test.ts]: accept

File: src/lib/setup/pipelineService.ts (hunk 6 above)
Comment (dennisseah) [reply src/lib/setup/pipelineService.ts]: accept

File: src/lib/setup/prompt.ts (hunk 7 above)
Comment (dennisseah) [reply src/lib/setup/prompt.ts]: accept

**Review by bnookala — APPROVED** [review]

**Pull request merged**